source/math/FloatPoint2DVectorList.ooc

```diff
@@ -107,4 +108,30 @@ FloatPoint2DVectorList: class extends VectorList<FloatPoint2D> {
 			result = result >> this[i] toString() >> "\n"
 		result
 	}
+	resampleLinear: func (start, end, interval: Float) -> This {
+		// Assumes list is sorted by x values.
+		resultCount := ((end - start) / interval) ceil() as Int + 1
+		result := This new()
+		previousIndex := 0
+		for (i in 0 .. resultCount) {
+			point := FloatPoint2D new(start + i * interval, 0.0f)
+			if (point x <= this[0] x)
```

Comment:
Well, I thought about it, but I think the difference would be negligible compared to the ugliness of the code, since I'm indexing at so many different indices. Also performance is not critical, so I think we can tackle it if we need it.